### ISSUE TITLE
LH warns of preconnect link not used by the browser when there is no preconnect in the code

### ISSUE THREAD

**Issue by antoinebisch** [GoogleChrome/lighthouse#10970]
## Provide the steps to reproduce
1. This issue seems to be on and off across pages on the https://www.lacoste.com domain. At the time of writing, it appears on https://www.lacoste.com/fr/a-place-in-the-sun/ but it has also been observed on https://www.lacoste.com/fr/

#### What is the current behavior?

When running a LH report, the following warning appears (the file name highlighted varies across tests):  

![image](https://user-images.githubusercontent.com/38130294/84759145-3ff6b000-afbe-11ea-917e-cb9ab63b4001.png)

#### What is the expected behavior?

This warning should not appear as there is no preconnect used in the source code:

![image](https://user-images.githubusercontent.com/38130294/84759350-85b37880-afbe-11ea-8f3b-5235fb3ded89.png)

The resource in question is not appearing in the Performance tab either:

![image](https://user-images.githubusercontent.com/38130294/84759489-b4315380-afbe-11ea-9e40-a356fb5280e4.png)

#### Environment Information
* Lighthouse version: 5.7.1
* Chrome version: 83.0.4103.106
* Operating System: macOS Version 10.15.5 (Build 19F101)

**Comment by patrickhulce** [comment]
Thanks for filing @antoinebisch! I'm not able to reproduce this warning on that URL right now. In fact it's suggesting the page use preconnect for it.

Two notes:

1. This page *does* load resources from that origin. It just happens later in load.
![image](https://user-images.githubusercontent.com/2301202/84825371-f4a4c780-afe6-11ea-926e-7c413647cc23.png)
2. The screenshot of the `<link rel="alternate">` tags in the source of the root document is not really evidence of the link not existing. If a `<link rel="preconnect">` is injected by script into the document it would still be flagged.

Looking at the URL it looks like some sort of third-party for helping other third-parties "suit your visual, motor and cognitive needs". Is it possible something on the page is adding this preconnect without your knowledge?

**Comment by antoinebisch** [comment]
Hey @patrickhulce, thank you so much for picking this up and looking into the issue!

I have just ran 3 other reports and only got the warnings on the 3rd try. I now got the warning on different files, on the homepage this time (https://www.lacoste.com/fr/) see below:

![image](https://user-images.githubusercontent.com/38130294/84906741-3f3b4800-b0aa-11ea-93ff-6f7b8ef929c1.png)

You are right that the preconnect scripts could be injected at runtime, although searching through the Elements tab (which I think would show the dynamically injected preconnect tags), I can't see a trace of them either:

![image](https://user-images.githubusercontent.com/38130294/84906997-8f1a0f00-b0aa-11ea-9129-fc45abd57726.png)

I will do a further investigation on wether a script could be injecting preconnect tags in these pages at runtime but my current observation is that there is none. Could it be something that Chrome itself attempts to do by anticipation?

Thank you.

**Comment by patrickhulce** [comment]
> Could it be something that Chrome itself attempts to do by anticipation?

No, not Chrome itself. If Chrome ever were to implement such a preconnect scanner, it wouldn't inject actual DOM elements into the page which is how Lighthouse detects these. 

It's certainly possible there's a Chrome extension or other software local to your machine that is doing this injection though. The fact that it's intermittent definitely suggests that it's some third-party you don't know about doing this.

Friendly tip while trying to debug, you can match the exact behavior of lighthouse by scanning the output of something like the below in the console instead :)

```js
[...document.querySelectorAll('link')].map(link => [link, link.rel, link.href])
```

**Comment by antoinebisch** [comment]
Hi Patrick,

Thanks for the console tip, very useful.

We found out in the end that it came from Akamai which anticipates and creates preconnects on the fly.

Do you think this is something that Lighthouse could potentially understand and pinpoint where these manipulations come from? This could be a more useful warning.

Thank you.

**Comment by connorjclark** [comment]
Is this a client side library?

**Comment by antoinebisch** [comment]
Server side injected, through the headers, which explains why it did not appear in the source:

![Capture](https://user-images.githubusercontent.com/38130294/85156015-64b68600-b251-11ea-968c-9139f829f774.JPG)

**Comment by rockeynebhwani** [comment]
@antoinebisch  - I see 7 preconnects on https://www.lacoste.com/gb/. You may want to check these as recommendation is not to have more than 2 or 3 pre-connects. 

![image](https://user-images.githubusercontent.com/1718757/95793279-21d75d00-0cdd-11eb-93cc-38ea6f6590ba.png)

**Comment by connorjclark** [comment]
We added a preconnect limit in v6.0

**Comment by connorjclark** [comment]
Let's consider listing all preconnects in the audit (even if passing), and showing warnings as subitems. Also could make a new warning detail type.

Or we could make a new audit for this.

ref #11960

**Comment by patrickhulce** [comment]
We've decided we'll merge this line of work into #11960. As warnings there are pretty low limits to what we can accomplish but as a separate audit, the message is stronger about not preconnecting to too many things.
